messaging/WFLY-10518_artemis_no_local_broker.adoc

```diff
@@ -0,0 +1,166 @@
+ broker will no longer require the presence of a local instance of Artemis.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10518[WFLY-10518]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1014[EAP7-1014]
+* https://issues.jboss.org/browse/EAP7-1006[EAP7-1006] is a RFE related to Java EE7 JMS resource
+definitions that will benefit from this RFE.
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mnovak@redhat.com[Miroslav Novak]
+
+=== Affected Projects or Components
+
+* This is a internal restructuring of the `messaging-activemq` subsystem that requires no changes
+from Artemis or other messaging components.
+
+=== Other Interested Projects
+
+== Requirements
+
+* It must be possible to configure WildFly to connect to a remote Artemis-based broker (using the Artemis resource adapter)
+  without having to run a local Artemis broker in WildFly.
+* Use Capabilities/Requirements to ensure that Artemis client or server resources are properly configured.
+* Previous and current subsystem schemas and models must remain compatible (using aliases to the new resources).
+
+=== Nice-to-Have Requirements
+
+* Refactor the Artemis modules to have a proper split between Artemis client and server code.
+* The Artemis resource adapter should only need to depend on Artemis client module.
+
+== Proposed Design
+
+Move resources up from `/subsystem=messaging/server` to `/subsystem=messaging/`:
+
+  * `connector`, `http-connector`, `remote-connector`, `in-vm-connector`
+  * `discovery-group`
+  * `pooled-connection-factory`
+  * `connection-factory`
+
+Note that a `server` can still be able to reference some of these resources
+(e.g. `discovery-group`).
+
+Other resources that remains in `/subsystem=messaging/server`:
+
+  * `acceptor`, `http-acceptor`, `remote-acceptor`, `in-vm-acceptor` (they corresponds to Artemis broker configuration)
+  * `broadcast-group` (this kind of resource is only used by Artemis brokers to broadcast cluster information)
+
+=== JMS Destination resources
+
+With the proposed design, a `pooled-connection-factory` resource is able to connect to a
+remote Artemis broker.
+However, to provide a fully functional JMS application, the user also needs to inject (or
+  lookup in JNDI) JMS Destinations object that represents the queue or topic managed by the remote
+  broker.
+
+There are `jms-queue` and `jms-topic` resources under the `server` resources but they require
+broker code to create and configure the queue on the local Artemis broker.
+
+We need another type of resources that represents JMS Destination object to store in JNDI.
+
+These new resources will be under `/subsystem=messaging-activemq`:
+
+  * `client-jms-queue`
+  * `client-jms-topic`
+
+These resource have only 2 attributes:
+
+  * `name` - the name of the destination that *MUST* be identical to the name of the destinations configured
+    in the remote Artemis broker
```

Comment:
How is a 'name' attribute distinct from the value of the last element in the resource address?

Reply:
sorry, the description is not correct. 
What I meant is that the `name` of the resource is meaningful and must be identical to the name of the destination in the remote Artemis broker.